### PR TITLE
Tests refactoring

### PR DESCRIPTION
**Pull request by Soullivaneuh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A

This test refactoring goal is to simplify test creation by managing the common tasks.

In a nutshell, a developper who creates a class which fully match `IsoCodeInterface`
will only have to provide test data by inheriting `AbstractIsoCodeInterfaceTest`.

In any case, each test classes must inherit `AbstractIsoCodeTest` to have common tests working.

Changes list:

* Created `AbstractIsoCodeTest` class with `getValidValues` and `getInvalidValues`
* Added `AbstractIsoCodeTest::testEmptyValues` to test empty values. No need to provide them anymore.
* Created `AbstractIsoCodesInterfaceTest` class with `testValidValues` and `testInvalidValues`. No need to provide test logic anymore.
* Refactored all test classes based on new abstracts.
* Created `PhoneNumberTest` common class with all data sets. Other related class are deleted.
* Created `ZipCodeTest` common class with all data sets. Other related class are deleted.
* Refactored `IPTest` to make it compatible with `AbstractIsoCodeTest`.
* Tweaked `PhoneNumber` class to allow empty second parameter.
* Tweaked `ZipCode` class to allow empty second parameter. This returns `false` if `null`.
* Fixed a typo on `SwiftBicTest` class name.

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
@ronanguilloux any chance to have a quick review here? I would like to submit another validator but I'm waiting the issue of this PR before. :-)

**Comment by Soullivaneuh** [comment]
Branch rebased to avoid conflict.

File: tests/IsoCodes/Tests/AbstractIsoCodeTest.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * @return array[]
+     */
+    abstract public function getValidValues();
+
+    /**
+     * @return array[]
+     */
+    abstract public function getInvalidValues();
+
+    final public function testEmptyValues()
```
Comment (ronanguilloux) [inline tests/IsoCodes/Tests/AbstractIsoCodeTest.php]: This method is never used.

Comment (Soullivaneuh) [inline tests/IsoCodes/Tests/AbstractIsoCodeTest.php]: Wrong, this is a `test*` method, so each test class extending `AbstractIsoCodeTest` will do the `testEmptyValues` on it.

Just put a `die` inside the function, you will see! :+1:

Comment (Soullivaneuh) [inline tests/IsoCodes/Tests/AbstractIsoCodeTest.php]: The goal of this method is to be sure to test empty values on each validator without duplicate provided data on each validator test class.

**Comment by Soullivaneuh** [comment]
Thanks a lot @ronanguilloux!

**Comment by ronanguilloux** [comment]
Thank you

**Pull request merged**